src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -493,6 +519,108 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
     }
   }
 
+  /**
+   * Handle the noPlayableRepresentation event, trigger an error if no fallback is possible.
+   * and can trigger event "noPlayableTrack"
+   * @param period - The period that has no playable representation
+   * @param trackType - The media type that is not playable
+   */
+  private onNoPlayableRepresentation(period: IPeriodMetadata, trackType: ITrackType) {
+    const periodHasAdaptationForType =
+      period.adaptations[trackType] !== undefined &&
+      period.adaptations[trackType].length > 0;
+    if (!periodHasAdaptationForType) {
+      log.debug(
+        `TS: The period does not have adaptation for ${trackType} there is no track to choose`,
+      );
+      return;
+    }
+
+    const firstPlayableAdaptation = findFirstPlayableAdaptation(period, trackType);
+    if (
+      firstPlayableAdaptation === undefined &&
+      (trackType === "text" || this.onTracksNotPlayableForType[trackType] === "continue")
+    ) {
+      // audio or video is not playable, but let's continue the playback without audio
+      // or without video because of the option was set to "continue".
+      log.warn(`TS: No playable ${trackType}, continuing without ${trackType}`);
+      this.trigger("noPlayableTrack", {
+        trackType,
+        period: { id: period.id, start: period.start, end: period.end },
+      });
+      // The previous event trigger could have had side-effects, so we
+      // re-check if we're still mostly in the same state
+      if (this._isDisposed) {
+        return; // Someone disposed the `TracksStore` on the previous side-effect
+      }
+    } else if (firstPlayableAdaptation === undefined) {
+      const noRepErr = new MediaError(
+        "NO_PLAYABLE_REPRESENTATION",
+        `No ${trackType} Representation can be played`,
+        { tracks: undefined },
+      );
```

Comment:
So this event now is also sent when the codec is not supported? That's a big change

Reply:
Not for when the codec wasn't supported which now can happen due to the `"continue"` part

Reply:
In that case, I'm wondering if a `MANIFEST_INCOMPATIBLE_CODECS_ERROR` error wouldn't be preferable